src/app/loadout-drawer/loadout-drawer-reducer.ts

```diff
@@ -194,64 +213,80 @@ export function removeItem(
   });
 }
 
+function setEquipForItemInLoadout(
+  defs: D1ManifestDefinitions | D2ManifestDefinitions,
+  item: DimItem,
+  loadout: Loadout,
+  isEquipped: boolean
+) {
+  // Subclasses and some others are always equipped
+  if (singularBucketHashes.includes(item.bucket.hash)) {
+    return;
+  }
+
+  // TODO: it might be nice if we just assigned a unique ID to every loadout item just for in-memory ops like deleting
+  // We can't just look it up by identity since Immer wraps objects in a proxy and getItemsFromLoadoutItems
+  // changes the socketOverrides, so simply search by unmodified ID and hash.
+  const loadoutItemIndex = findSameLoadoutItemIndex(
+    defs,
+    loadout.items,
+    convertToLoadoutItem(item, false, 1)
+  );
+
+  if (loadoutItemIndex === -1) {
+    return;
+  }
+  const loadoutItem = loadout.items[loadoutItemIndex];
+
+  if (item.equipment) {
+    if (!isEquipped) {
+      // It's equipped, mark it unequipped
+      loadoutItem.equip = false;
```

Comment:
Looks like this is mutating the loadout. Should it explicitly operate on `Draft<Loadout>`?

Reply:
I think that is a good idea, hadn't thought about that.

Reply:
Done